presto-spi/src/main/java/com/facebook/presto/spi/plan/OrderingScheme.java

```diff
@@ -11,76 +11,75 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.facebook.presto.sql.planner;
+package com.facebook.presto.spi.plan;
 
 import com.facebook.presto.spi.block.SortOrder;
 import com.facebook.presto.spi.relation.VariableReferenceExpression;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.ImmutableSet;
-import com.google.common.collect.Maps;
 
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
 
-import static com.google.common.base.MoreObjects.toStringHelper;
-import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.collect.ImmutableList.toImmutableList;
-import static com.google.common.collect.Iterables.getOnlyElement;
 import static java.lang.String.format;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Collections.unmodifiableSet;
 import static java.util.Objects.requireNonNull;
 
 public class OrderingScheme
 {
     private final List<VariableReferenceExpression> orderBy;
     private final Map<VariableReferenceExpression, SortOrder> orderings;
 
+    /**
+     * OrderingScheme indicates the sort order of rows:
+     * `orderby`:  keys to sort
+     * `orderings`: sort directions of keys
+     */
     @JsonCreator
-    public OrderingScheme(@JsonProperty("orderBy") List<VariableReferenceExpression> orderBy, @JsonProperty("orderings") Map<VariableReferenceExpression, SortOrder> orderings)
+    public OrderingScheme(
+            @JsonProperty("orderBy") List<VariableReferenceExpression> orderBy,
+            @JsonProperty("orderings") Map<VariableReferenceExpression, SortOrder> orderings)
     {
         requireNonNull(orderBy, "orderBy is null");
         requireNonNull(orderings, "orderings is null");
         checkArgument(!orderBy.isEmpty(), "orderBy is empty");
-        checkArgument(orderings.keySet().equals(ImmutableSet.copyOf(orderBy)), "orderBy keys and orderings don't match");
-        this.orderBy = ImmutableList.copyOf(orderBy);
-        this.orderings = ImmutableMap.copyOf(orderings);
+        checkArgument(orderings.keySet().equals(immutableSetCopyOf(orderBy)), "orderBy keys and orderings don't match");
+        this.orderBy = immutableListCopyOf(orderBy);
+        this.orderings = immutableMapCopyOf(orderings);
     }
 
+    /**
+     *  Sorting keys of rows.
```

Comment:
I don't really understand the comment (the function name `getOrderBy` is more explanatory than this comment).

Reply:
Would the following be a better comment: `List of keys to order by`?

Reply:
We used to use `symbol`, now we use `variable`, but `key` was never a defined concept. So I don't really know what this word is referring to. I might just say `List of order by variables`, which is basically what the method signature says so I might just remove this. 😂